GenericsHomework/GenericsHomework/Node.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System.Collections;
+
+namespace GenericsHomework;
+
+public class Node<T> : ICollection<T>, IEnumerable<T>
+{
+    public T Data
+    {
+        get => _Data!;
+        set => _Data = value ?? throw new ArgumentNullException(nameof(Data));
+    }
+    private T? _Data;
+
+    public Node<T> Next
+    {
+        get => _Next!;
+        private set => _Next = value ?? throw new ArgumentNullException(nameof(Next));
+    }
+
+    private int _Count;
+    public int Count => _Count!;

```

Comment:
```suggestion
    public int Count => _Count;
```
IDE Message: "The nullable warning suppression expression is redundant"